riscv-asm.md

```diff
@@ -26,19 +26,39 @@ and vector registers (V extension).
 
 ## General registers
 
-The RV32I base integer ISA includes 32 registers, named `x0` to `x31`. The
-program counter `PC` is separate from these registers, in contrast to other
-processors such as the ARM-32. The first register, `x0`, has a special function:
-Reading it always returns 0 and writes to it are ignored. As we will see later,
-this allows various tricks and simplifications.
+The **instruction set architecture** (ISA) of RISC-V contains 32 registers.
```

Comment:
The old wording was more precise.  It's a true statement that the RV32I and RV64I base ISAs contain 32 registers.  But the other base ISA (RV32E) has only 16 registers, and various ISA extensions (like F) increase the register count beyond 32.